src/Microsoft.ML.Transforms/Text/NgramHashingTransformer.cs

```diff
@@ -1028,12 +970,193 @@ public VBuffer<ReadOnlyMemory<char>>[] SlotNamesMetadata(out ColumnType[] types)
                     if (_iinfoToCollector[iinfo] != null)
                     {
                         var vec = values[iinfo] = _iinfoToCollector[iinfo].GetMetadata();
-                        Contracts.Assert(vec.Length == _parent._bindings.Types[iinfo].VectorSize);
+                        Contracts.Assert(vec.Length == 1 << _parent._columns[iinfo].HashBits);
                         types[iinfo] = new VectorType(TextType.Instance, vec.Length);
                     }
                 }
                 return values;
             }
         }
     }
+
+    /// <summary>
+    /// Produces a bag of counts of ngrams (sequences of consecutive words of length 1-n) in a given text.
+    /// It does so by hashing each ngram and using the hash value as the index in the bag.
+    ///
+    /// <see cref="NgramHashingEstimator"/> is different from <see cref="WordHashBagEstimator"/> in a way that <see cref="NgramHashingEstimator"/>
+    /// takes tokenized text as input while <see cref="WordHashBagEstimator"/> tokenizes text internally.
+    /// </summary>
+    public sealed class NgramHashingEstimator : IEstimator<NgramHashingTransformer>
+    {
+        internal static class Defaults
+        {
+            internal const int NgramLength = 2;
+            internal const bool AllLengths = true;
+            internal const int SkipLength = 0;
+            internal const int HashBits = 16;
+            internal const uint Seed = 314489979;
+            internal const bool RehashUnigrams = false;
+            internal const bool Ordered = true;
+            internal const int InvertHash = 0;
+        }
+
+        private readonly IHost _host;
+        private readonly NgramHashingTransformer.ColumnInfo[] _columns;
+
+        /// <summary>
+        /// Produces a bag of counts of hashed ngrams in <paramref name="inputColumn"/>
+        /// and outputs ngram vector as <paramref name="outputColumn"/>
+        ///
+        /// <see cref="NgramHashingEstimator"/> is different from <see cref="WordHashBagEstimator"/> in a way that <see cref="NgramHashingEstimator"/>
+        /// takes tokenized text as input while <see cref="WordHashBagEstimator"/> tokenizes text internally.
+        /// </summary>
+        /// <param name="env">The environment.</param>
+        /// <param name="inputColumn">The column containing text to compute bag of word vector.</param>
```

Comment:
>The column containing text to compute bag of word vector. [](start = 38, length = 57)

I would change the comment to something like: 
Name of input column containing tokenized text.